### DISCUSSION TITLE
No way to pass variables to widgets?

### DISCUSSION THREAD

**Discussion by fuzzyhope1502** [filamentphp/filament#5018] [Help] [answered]
I have a UserResource and pages from Filament, and I've made a widget table that will display the current user's subscriptions and extra actions like add/remove.

Maybe I'm blind, but I've looked through docs multiple times and can find 0 reference on how to pass a variable/record to a widget. I attempted to pass it via __construct, but the first variable of that appears to be the livewire ID, and also I can't really specify parameters inside `getFooterWidgets` from what I can see. All I want to do is provide the user id of the current page user (Not Auth'd user), and scope the eloquent query to it. Thought this would be somewhat basic to achieve but guess not.

Is there a correct way to do this? At the moment I do something like this.
```
$userId = request()->route()->parameter('record');
```
Really not happy with this, it's really hacky and it's also leading to issues when trying to do modal confirmation (upon triggering an action on table, all table records disappear).

**Comment by danharrin** [comment, marked as answer]
https://filamentphp.com/docs/2.x/admin/resources/widgets#accessing-the-current-record

Widgets are Livewire components, so info persisted between requests must be stored in public properties

**Reply by synsyst** [reply]
Does this still count for v3? I have trouble getting a simple filter value from my table into a widget on my list page

**Reply by danharrin** [reply]
Like this? https://filamentphp.com/docs/3.x/panels/resources/widgets#accessing-page-table-data-in-the-widget

**Reply by synsyst** [reply]
Thanks for responding, it looks like that's what I need - Bit new to Filament and I spent my time hunting it in the widget section, I will try this out, thanks!
